File: packages/csharp/src/Anip.Core/BudgetContext.cs
using System.Text.Json.Serialization;

namespace Anip.Core;

public class BudgetContext
{
    [JsonPropertyName("budget_max")]
    public double BudgetMax { get; set; }

    [JsonPropertyName("budget_currency")]
    public string BudgetCurrency { get; set; } = "";

    [JsonPropertyName("cost_check_amount")]
    public double? CostCheckAmount { get; set; }

    [JsonPropertyName("cost_certainty")]
    public string? CostCertainty { get; set; }

    [JsonPropertyName("cost_actual")]
    public double? CostActual { get; set; }

    [JsonPropertyName("within_budget")]
    public bool WithinBudget { get; set; }
}

File: packages/java/anip-core/src/main/java/dev/anip/core/BudgetContext.java
package dev.anip.core;

/**
 * Provides budget evaluation details in invocation responses.
 */
public class BudgetContext {

    private double budgetMax;
    private String budgetCurrency;
    private Double costCheckAmount;
    private String costCertainty;
    private Double costActual;
    private boolean withinBudget;

    /** No-arg constructor for Jackson deserialization. */
    public BudgetContext() {}

    public BudgetContext(double budgetMax, String budgetCurrency,
                         Double costCheckAmount, String costCertainty,
                         Double costActual, boolean withinBudget) {
        this.budgetMax = budgetMax;
        this.budgetCurrency = budgetCurrency;
        this.costCheckAmount = costCheckAmount;
        this.costCertainty = costCertainty;
        this.costActual = costActual;
        this.withinBudget = withinBudget;
    }

    public double getBudgetMax() {
        return budgetMax;
    }

    public String getBudgetCurrency() {
        return budgetCurrency;
    }

    public Double getCostCheckAmount() {
        return costCheckAmount;
    }

    public String getCostCertainty() {
        return costCertainty;
    }

    public Double getCostActual() {
        return costActual;
    }

    public boolean isWithinBudget() {
        return withinBudget;
    }

    public void setBudgetMax(double budgetMax) {
        this.budgetMax = budgetMax;
    }

    public void setBudgetCurrency(String budgetCurrency) {
        this.budgetCurrency = budgetCurrency;
    }

    public void setCostCheckAmount(Double costCheckAmount) {
        this.costCheckAmount = costCheckAmount;
    }

    public void setCostCertainty(String costCertainty) {
        this.costCertainty = costCertainty;
    }

    public void setCostActual(Double costActual) {
        this.costActual = costActual;
    }

    public void setWithinBudget(boolean withinBudget) {
        this.withinBudget = withinBudget;
    }
}
